sdcm/results_analyze/__init__.py

```diff
@@ -96,9 +96,21 @@ def _test_version(self, test_doc):
                 key = test_doc['_source']['versions'].get(value)
                 if key:
                     return key
+        else:
+            self.log.error('Scylla version is not found for test %s', test_doc['_id'])
+            return ''
+
+        res = dict()
+        if 'version' in test_doc['_source']['versions']:
+            res['version'] = test_doc['_source']['versions']['version']
+        if 'date' in test_doc['_source']['versions']:
+            res['date'] = test_doc['_source']['versions']['date']
+        if 'commit_id' in test_doc['_source']['versions']:
+            res['commit_id'] = test_doc['_source']['versions']['commit_id']
+        if 'build_id' in test_doc['_source']['versions']:
+            res['build_id'] = test_doc['_source']['versions']['build_id']
 
```

Comment:
why not?

Reply:
Above you have the if ... else. And both return value

Reply:
There is no answer, so this can't be marked as resolved...

Reply:
so, if we do enter this:
`if test_doc['_source'].get('versions'):`
then we search `for value in ('scylla-server', 'scylla-enterprise-server'):`

and then if `key = test_doc['_source']['versions'].get(value)` is not in `versions` then we will not return, and it will continue to what is here line 103